include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -1800,10 +1800,10 @@ ::FEEvaluationBase (const Mapping<dim>       &mapping,
                     const unsigned int        first_selected_component,
                     const FEEvaluationBase<dim,n_components_other,Number> *other)
   :
-  quad_no            (-1),
+  quad_no            (static_cast<unsigned int>(-1)),
   n_fe_components    (n_components_),
-  active_fe_index    (-1),
-  active_quad_index  (-1),
+  active_fe_index    (static_cast<unsigned int>(-1)),
+  active_quad_index  (static_cast<unsigned int>(-1)),
```

Comment:
@kronbichler -- shouldn't all of these be `numbers::invalid_unsigned_int`? Or do you somewhere test them against `-1`?